homepage/homepage/app/(docs)/docs/[framework]/[...slug]/using-covalues/cofeeds.mdx

```diff
@@ -0,0 +1,219 @@
+
+CoFeeds are defined by specifying the type of items they'll contain, similar to how you define CoLists:
+
+<CodeGroup>
+```ts
+// Define a schema for feed items
+class Activity extends CoMap {
+  timestamp = co.Date;
+  action = co.literal("watering", "planting", "harvesting", "maintenance");
+  notes = co.optional.string;
+}
+
+// Define a feed of garden activities
+class ActivityFeed extends CoFeed.Of(co.ref(Activity)) {}
+
+// Create a feed instance
+const activityFeed = ActivityFeed.create([]);
+```
+</CodeGroup>
+
+Like other CoValues, you can specify [ownership](/docs/using-covalues/ownership) when creating CoFeeds.
+
+## Reading from CoFeeds
+
+Since CoFeeds are made of entries from users over multiple sessions, you can access entries in different ways - from a specific user's session or from their account as a whole.
+
+### Per-Session Access
+
+To retrieve entries from a session:
+
+<CodeGroup>
+```ts
+// Get the feed for a specific session
+const sessionFeed = activityFeed.perSession[sessionId];
+
+// Latest entry from a session
+console.log(sessionFeed.value.action); // "watering"
+```
+</CodeGroup>
+
+For convenience, you can also access the latest entry from the current session with `inCurrentSession`:
+
+<CodeGroup>
+```ts
+// Get the feed for the current session
+const currentSessionFeed = activityFeed.inCurrentSession;
+
+// Latest entry from the current session
+console.log(currentSessionFeed.value.action); // "harvesting"
+```
+</CodeGroup>
+
+### Per-Account Access
+
+To retrieve entries from a specific account you can use bracket notation with the account ID:
+
+<CodeGroup>
+```ts
+// Get the feed for a specific account
+const accountFeed = activityFeed[accountId];
+
+// Latest entry from the account
+console.log(accountFeed.value.action); // "watering"
+```
+</CodeGroup>
+
+For convenience, you can also access the latest entry from the current account with `byMe`:
+
+<CodeGroup>
+```ts
+// Get the feed for the current account
+const myLatestEntry = activityFeed.byMe;
+
+// Latest entry from the current account
+console.log(myLatestEntry.value.action); // "harvesting"
+```
+</CodeGroup>
+
+### Feed Entries
+
+#### All Entries
+
+To retrieve all entries from a CoFeed:
+
+<CodeGroup>
+```ts
+// Get the feeds for a specific account and session
+const accountFeed = activityFeed[accountId];
+const sessionFeed = activityFeed.perSession[sessionId];
+
+// Iterate over all entries from the account
+for (const entry of accountFeed.all) {
+  console.log(entry.value);
+}
+
+// Iterate over all entries from the session
+for (const entry of sessionFeed.all) {
+  console.log(entry.value);
+}
+```
+</CodeGroup>
+
+#### Latest Entry
+
+To retrieve the latest entry from a CoFeed, ie. the last update:
+
+<CodeGroup>
+```ts
+// Get the latest entry from a specific account
+const latestEntry = activityFeed.byMe;
+
+console.log(`My last action was ${latestEntry.value.action}`); 
+  // "My last action was harvesting"
+```
+</CodeGroup>
+
```

Comment:
What do you think about adding an example on how to list the last values from every account:

```ts
Object.values(activityFeed).map(coFeedAccountEntry => ({
  accountName: coFeedAccountEntry.by?.profile?.name,
  value: coFeedAccountEntry.value,
}))
```

Was about to propose another example on how to list all values of the entire CoFeed but the code is just too horrible to be in the docs.

```ts
  Object.values(activityFeed).flatMap(coFeedAccountEntry => Array.from(coFeedAccountEntry.all, entry => ({
  accountName: entry.by?.profile?.name,
  value: entry.value,
})))
```

Reply:
I've added it. I'll add it to the Common Patterns too (in another PR).

And yes, not adding your 2nd one... we're not trying to scare folks off...